### PR TITLE
ci: add humble, remove galactic, update rolling

### PR DESCRIPTION
**Pull request by wep21**
- add humble
- remove galactic because of its eol
- update Ubuntu version for rolling
- fix mistaken matrix

### CODE REVIEW

**Comment by wep21** [comment]
@esteve Could you review this PR?

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -10,25 +10,21 @@ jobs:
...
+          # Humble Hawksbill (May 2022 - May 2027)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-jammy-ros-humble-ros-base-latest
+            ros_distribution: humble
             ros_version: 2
           # Rolling Ridley  (June 2020 - Present)
-          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-rolling-ros-base-latest
+          - docker_image: rostooling/setup-ros-docker:ubuntu-jammy-ros-rolling-ros-base-latest
             ros_distribution: rolling
             ros_version: 2
-        linter:
```
Comment (esteve) [inline .github/workflows/build_test.yml:28]: Any reason for removing the lint checks?

**Comment by esteve** [comment]
@wep21 thanks. Only one question about the removal of the linter checks, but the rest looks fine.

File: .github/workflows/build_test.yml (hunk 1 above)
Comment (wep21) [reply .github/workflows/build_test.yml:28]: I think linters should be passed to `action-ros-lint` like [this](https://github.com/ros-tooling/topic_tools/blob/fdc8df8361fd449c4e6564a4fdb04ecbd6771870/.github/workflows/lint.yml#L28-L30), but currently they are never used, so the same ci is executed 3 times wrongly.
https://github.com/ros-drivers/transport_drivers/actions/runs/4049784318/jobs/7144340139
https://github.com/ros-drivers/transport_drivers/actions/runs/4049784318/jobs/7144340221
https://github.com/ros-drivers/transport_drivers/actions/runs/4049784318/jobs/7144340325

Comment (wep21) [reply .github/workflows/build_test.yml:28]: @esteve friendly ping

Comment (esteve) [reply .github/workflows/build_test.yml:28]: @wep21 sorry, was away for a few days. Thanks for the explanation

**Review by esteve — APPROVED** [review]

**Pull request merged**